src/components/dialogs/SignUpDialog.svelte

```diff
@@ -0,0 +1,107 @@
+ -->
+
+<script lang="ts">
+  import { t } from '../../i18n';
+  import { hasSeenSignUpDialog } from '../../script/stores/uiStore';
+  import StandardDialog from './StandardDialog.svelte';
+  import StandardButton from '../StandardButton.svelte';
+  import { signUp as signUpApi } from '../../script/utils/api';
+  import HtmlFormattedMessage, { linkWithProps } from '../HtmlFormattedMessage.svelte';
+  import LoadingAnimation from '../LoadingSpinner.svelte';
+
+  export let isOpen: boolean;
+  let email = '';
+  let emailInput: HTMLInputElement;
+  let emailIsValid: boolean | undefined;
+  let signUpError = false;
+  let submissionInProgress = false;
+
+  const onClose = () => {
+    $hasSeenSignUpDialog = true;
+    isOpen = false;
+  };
+
+  const handleSignup = async (e: Event) => {
+    e.preventDefault();
+    submissionInProgress = true;
+    signUpError = false;
+    emailIsValid = emailInput.validity.valid;
+    if (!emailIsValid) {
+      submissionInProgress = false;
+      return;
+    }
+    const result = await signUpApi(email);
+    if (result) {
+      onClose();
+    } else {
+      signUpError = true;
+    }
+    submissionInProgress = false;
+  };
+</script>
+
+<StandardDialog {isOpen} {onClose} class="w-150 space-y-5">
+  <svelte:fragment slot="heading">{$t('sign-up.title')}</svelte:fragment>
+  <svelte:fragment slot="body">
+    <div class="space-y-3">
+      <p>{$t('sign-up.content')}</p>
+      <form class="space-y-2" on:submit={handleSignup} novalidate>
+        <label class="font-bold" for="email">
+          {$t('sign-up.email-label')} <span class="text-red-600" aria-hidden>*</span>
+        </label>
+        <input
+          class="block w-4/5 h-10 border rounded-md border-gray-300 border-solid px-4 outline-none focus-visible:ring-4 focus-visible:ring-offset-1 focus-visible:ring-ring"
+          class:!border-red-600={emailIsValid === false}
+          id="email"
+          name="email"
+          bind:value={email}
+          bind:this={emailInput}
+          disabled={submissionInProgress}
+          type="email"
+          required />
+        {#if emailIsValid === false}
+          <p class="text-red-600" aria-live="polite">
+            {$t('sign-up.invalid')}
+          </p>
+        {/if}
+        {#if signUpError}
+          <p class="text-red-500" aria-live="polite">
+            <HtmlFormattedMessage
+              id="sign-up.error"
+              options={{
+                values: {
+                  link: linkWithProps(
+                    {
+                      href: 'mailto:thenextgen.help@microbit.org',
+                    },
+                    {
+                      className: 'text-red-500 underline',
+                    },
+                  ),
+                },
+              }} />
+          </p>
+        {/if}
+      </form>
+    </div>
+    <div class="flex items-center justify-end gap-x-5">
+      <StandardButton onClick={onClose}>{$t('sign-up.skip-action')}</StandardButton>
```

Comment:
To confirm whether this is a link-like button in the bottom left or a standard secondary-styled button.

Reply:
What it might look like? Was doing stuff on Figma, so thought I'd mock it up as well
<img width="549" alt="Screenshot 2024-03-06 at 16 34 59" src="https://github.com/microbit-foundation/ml-trainer/assets/145345672/4c44887a-5460-42a4-a6e6-cab6b9cd915a">